### PR TITLE
redo global cooldown to be its own value

### LINKED ISSUE #1906: enhancement: Unstupify Global Cooldowns
### Description

Global cooldowns are stupid and I should feel bad.
I set them to simply loop through EVERY object in the item/spell database and add a cooldown value.
That was a very bad idea.

I should not have done that.

### Intended Use-Case

Make global cooldowns great again.

### Duplicate Check

- [X] This enhancement idea is not a duplicate to the best of my knowledge.

### PR DESCRIPTION
**Pull request by Cheshire92**
Resolves #1906 

I am a big dum.

### CODE REVIEW

File: Intersect.Client/Entities/Player.cs [hunk 1]
```diff
@@ -1049,10 +1066,14 @@ public void TryUseSpell(int index)
 
         public long GetSpellCooldown(Guid id)
         {
-            if (SpellCooldowns.ContainsKey(id))
+            if (SpellCooldowns.ContainsKey(id) && SpellCooldowns[id] > Timing.Global.Milliseconds)
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:1069]: Use TryGet && comparison to avoid double lookup

File: Intersect.Client/Entities/Player.cs [hunk 2]
```diff
@@ -1049,10 +1066,14 @@ public void TryUseSpell(int index)
...
             {
                 return SpellCooldowns[id];
             }
+            else if (!SpellBase.Get(id).IgnoreGlobalCooldown && Globals.Me.GlobalCooldown > Timing.Global.Milliseconds)
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:1073]: Also use TryGet here, Get can return null and we'd have a null pointer exception here

File: Intersect.Client/Entities/Player.cs [hunk 3]
```diff
@@ -577,6 +591,10 @@ public long GetSpellRemainingCooldown(int slot)
                     {
                         return ItemCooldowns[spl.Id] - Timing.Global.Milliseconds;
                     }
+                    else if (!SpellBase.Get(spl.Id).IgnoreGlobalCooldown && Globals.Me.GlobalCooldown > Timing.Global.Milliseconds)
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:594]: TryGet to avoid NPE

File: Intersect.Client/Entities/Player.cs [hunk 4]
```diff
@@ -560,6 +570,10 @@ public bool IsSpellOnCooldown(int slot)
                     {
                         return true;
                     }
+                    else if (!SpellBase.Get(spl.Id).IgnoreGlobalCooldown && Globals.Me.GlobalCooldown > Timing.Global.Milliseconds)
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:573]: TryGet to avoid NPE

File: Intersect.Client/Entities/Player.cs [hunk 5]
```diff
@@ -543,6 +549,10 @@ public long GetItemRemainingCooldown(int slot)
                     {
                         return ItemCooldowns[itm.ItemId] - Timing.Global.Milliseconds;
                     }
+                    else if (!ItemBase.Get(itm.ItemId).IgnoreGlobalCooldown && Globals.Me.GlobalCooldown > Timing.Global.Milliseconds)
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:552]: TryGet to avoid NPE

File: Intersect.Client/Entities/Player.cs [hunk 6]
```diff
@@ -526,6 +528,10 @@ public bool IsItemOnCooldown(int slot)
                     {
                         return true;
                     }
+                    else if (!ItemBase.Get(itm.ItemId).IgnoreGlobalCooldown && Globals.Me.GlobalCooldown > Timing.Global.Milliseconds)
```
Comment (lodicolo) [inline Intersect.Client/Entities/Player.cs:531]: TryGet to avoid NPE

**Review by lodicolo — CHANGES_REQUESTED** [review]

**Comment by Cheshire92** [comment]
Right, I uh.. I saw that while I was writing it and neglected to fix it when I figured out what was messing with my timers. lmao

**Review by lodicolo — APPROVED** [review]

**Pull request merged**